tests/sqlite3/test-sqlite3.cpp

```diff
@@ -53,6 +53,62 @@ TEST_CASE("SQLite rowid", "[sqlite][rowid][oid]")
     sql << "drop table test1";
 }
 
+static void SetupForeignKeysTest(soci::session& sql)
+{
+    sql <<
+    "create table parent ("
+    "    id integer primary key"
+    ")";
+
+    sql <<
+    "create table child ("
+    "    id integer primary key,"
+    "    parent integer,"
+    "    foreign key(parent) references parent(id)"
+    ")";
+
+    sql << "insert into parent(id) values(1)";
+    sql << "insert into child(id, parent) values(100, 1)";
+}
+
+TEST_CASE("SQLite foreign keys are disabled by default", "[sqlite][foreignkeys]")
+{
+    soci::session sql(backEnd, connectString);
+
+    SetupForeignKeysTest(sql);
+
+    sql << "delete from parent where id = 1";
+
+    int parent = 0;
+    sql << "select parent from child where id = 100 ", into(parent);
+
+    CHECK(parent == 1);
+
+    sql << "drop table child";
+    sql << "drop table parent";
+}
+
+TEST_CASE("SQLite foreign keys are enabled by foreign_keys option", "[sqlite][foreignkeys]")
+{
+    soci::session sql(backEnd, "dbname=:memory: foreign_keys=on");
+
+    SetupForeignKeysTest(sql);
+
+    bool thrown = false;
+    try
+    {
+        sql << "delete from parent where id = 1";
+    }
+    catch (const soci::soci_error&)
+    {
+        thrown = true;
+    }
+    CHECK(thrown);
```

Comment:
This should be replaced with just `CHECK_THROWS_AS()` or, if possible, with `CHECK_THROWS_WITH` to also verify that the _expected_ exception is thrown and not something else.